cpp/src/groupby/hash/groupby.cu

```diff
@@ -421,36 +425,32 @@ void sparse_to_dense_results(table_view const& keys,
  * @brief Construct hash map that uses row comparator and row hasher on
  * `d_keys` table and stores indices
  */
-auto create_hash_map(table_device_view const& d_keys,
-                     bool keys_have_nulls,
-                     null_policy include_null_keys,
-                     rmm::cuda_stream_view stream)
+auto create_hash_map(
+  std::shared_ptr<cudf::experimental::row::hash::preprocessed_table> const& preprocessed_keys,
```

Comment:
Taking a `shared_ptr<T> const&` is a bit ambiguous as it declares "this function _may_ take partial ownership of the object".

Passing by value is preferred when possible as it clearer "this function _will_ take partial ownership of the managed object". 
```suggestion
  std::shared_ptr<cudf::experimental::row::hash::preprocessed_table> preprocessed_keys,
```

Reply:
Done